engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/StaticAnalysisTest.java

```diff
@@ -0,0 +1,117 @@
+package org.enso.compiler.test;
+
+import com.oracle.truffle.api.TruffleFile;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import org.enso.common.LanguageInfo;
+import org.enso.common.MethodNames;
+import org.enso.common.RuntimeOptions;
+import org.enso.compiler.core.ir.Module;
+import org.enso.editions.LibraryName;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.util.TruffleFileSystem;
+import org.enso.interpreter.test.InterpreterContext;
+import org.enso.pkg.Config;
+import org.enso.pkg.Contact;
+import org.enso.pkg.Package;
+import org.enso.pkg.QualifiedName;
+import org.graalvm.polyglot.Source;
+import scala.Option;
+import scala.jdk.javaapi.CollectionConverters;
+
+public abstract class StaticAnalysisTest {
+
+  /**
+   * The interpreter context is needed here as it ensures initialization of everything needed to
+   * perform imports resolution, including PackageRepository.
+   *
+   * <p>Ideally, the tests for the static analysis capabilities of the compiler should _not_ depend
+   * on the Graal runtime context, as they should be runnable in other contexts - i.e. in a Visual
+   * Studio Code language server.
```

Comment:
That would indeed be ideal! The integration of the `runtime-compiler` into VSCode is blocked right now because we have no way to run the `runtime-compiler` without `TruffleCompilerContext` right now. If only these tests could be _moved to `runtime-compiler` project test_ directory! Then:
- they wouldn't depend on `TruffleCompilerContext`
- they would need an _independent compiler context_ implementation
- such an implementation could be used in VSCode as well

I remember @Akirathan was once dreaming about using `runtime-compiler` in VSCode and this was a huge _"known unknown"_.

Reply:
Yes I think ideally we should move them there - it seems like the right direction.

But implementing the independent compiler context is definitely a task for a separate PR.